### PR TITLE
Add support for $nearSphere

### LINKED ISSUE #1438: Add nearSphere to MongodbExpressions.
Since `$near` cannot operate on a `2dsphere` index using the legacy coordinate format it would be helpful to have something like a `MongodbExpressions.nearSphere`.

```java
class MongodbOps {
  
  // ...
  public static final Operator<Boolean> NEARSPHERE = new OperatorImpl<Boolean>(NS, "NEARSPHERE");
}

class MongodbExpressions {
  
  // ...
  public static BooleanExpression nearSphere(Expression<Double[]> expr, double latVal, double longVal) {
    return BooleanOperation.create(MongodbOps.NEARSPHERE, expr, ConstantImpl.create(new Double[]{latVal, longVal}));
  }
}

class MongodbSerializer {
  
  public Object visit(Operation<?> expr, Void context) {
    
    // ...
    } else if (op == MongodbOps.NEARSPHERE) {
      return asDBObject(asDBKey(expr, 0), asDBObject("$nearSphere", asDBValue(expr, 1)));
    }
  }
}
```

It would also be nice to have support for `$minDistance` and `$maxDistance`.

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1438

### CODE REVIEW

**Comment by johnktims** [comment]
Looks good to me. Do you think we should add an integration test?

File: querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java [hunk 1]
```diff
@@ -256,6 +256,9 @@ public Object visit(Operation<?> expr, Void context) {
         } else if (op == MongodbOps.NEAR) {
             return asDBObject(asDBKey(expr, 0), asDBObject("$near", asDBValue(expr, 1)));
 
+        } else if (op == MongodbOps.NEARSPHERE) {
+            return asDBObject(asDBKey(expr, 0), asDBObject("$nearSphere", asDBValue(expr, 1)));
```
Comment (Shredder121) [inline querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java]: If `$elemMatch` becomes `ELEM_MATCH`, isn't it more consistent to use `NEAR_SPHERE` for `$nearSphere`?

**Comment by johnktims** [comment]
The updates look good to me. @Shredder121, feel free to merge it.

**Comment by Shredder121** [comment]
Yes it's now good, I'll merge it.

**Pull request merged**